core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/solver/change/ProblemChangeDirector.java

```diff
@@ -34,66 +34,78 @@
     /**
      * Add a new {@link PlanningEntity} instance into the {@link PlanningSolution working solution}.
      *
+     * @deprecated as not necessary for implementing {@link ProblemChange}s.
```

Comment:
I think that this requires much more explanation.
Why are those methods there, if you don't need them?
What happens if you call them?
How does a sample PFC look now?

All of that should, in my opinion, be explained in the class-level Javadoc.